kore/src/Kore/Internal/Substitution.hs

```diff
@@ -648,6 +649,73 @@ variables (NormalizedSubstitution subst) = Map.keysSet subst
 variables (Substitution subst) =
     foldMap (Set.singleton . assignedVariable) subst
 
+{- | Finds all the X = Y pairs and, whenever the provided function returns True
+for Y and False for X, it swaps the order to Y = X, but making sure the
+substitution is still normalized.
+-}
```

Comment:
```suggestion
{- | Apply an orientation to all variable-renaming substitutions.

A variable-renaming substitution is a pair of the form

@
X:S = Y:S
@

In a normalized substitution, the variable on the left-hand side of each substitution pair may not appear in any other substitution pair. The order of a variable-renaming pair is logically irrelevant, but often we have a preference for which of @X@ and @Y@ should appear on the left-hand side (that is, we have a preferred /orientation/).

@orientSubstitution@ applies an orientation to a normalized substitution and yields a normalized substitution. The orientation is expressed as a function

@
SomeVariableName variable -> Bool
@

returning `True` when the named variable is preferred on the left-hand side of a variable-renaming substitution pair. Each variable-renaming pair is oriented so that the variable on the left-hand side is a preferred variable, if possible. @orientSubstitution@ does not alter substitution pairs where both or neither variable is preferred for the left-hand side.
-}
```